fineract-provider/dependencies.gradle

```diff
@@ -31,83 +31,103 @@ dependencies {
+
+
+            [group: 'com.squareup.okhttp', name: 'okhttp', version: '2.0.0'],
+            [group: 'com.squareup.okhttp', name: 'okhttp-urlconnection', version: '2.0.0'],
+
+
+
+
+
+
+            //[group: 'net.sourceforge.javacsv', name: 'javacsv', version: '2.0'],
+            [group: 'org.apache.commons', name: 'commons-email', version: '1.3.3'],
+            [group: 'org.apache.commons', name: 'commons-lang3', version: '3.3.2'],
+
+            // no slf4j & logback here (anymore), as spring-boot-starter-logging already brings this now, better assembled (log4j-over-slf4j was originally forgotten here)
+
+            //[group: 'mysql', name: 'mysql-connector-java', version: '5.1.27'],
+            [group: 'org.drizzle.jdbc', name: 'drizzle-jdbc', version: '1.4'],
+            [group: 'org.apache.tomcat', name: 'tomcat-jdbc', version: tomcatVersion],
+
+
+            [group: 'org.apache.poi',name: 'poi', version: '3.9'],
+            [group: 'org.apache.poi',name: 'poi-ooxml', version: '3.9'],
+            [group: 'org.apache.poi',name: 'poi-ooxml-schemas', version: '3.9'],
+
+            [group: 'com.lowagie', name: 'itext', version: '2.1.7'],
+            //[group: 'com.lowagie', name: 'itext-rtf', version: '2.1.7'],
+
+            [group: 'com.googlecode.flyway', name: 'flyway-core', version: '2.1.1'],
+
+            [group: 'com.amazonaws', name: 'aws-java-sdk-s3', version: '1.11.80'],
+            [group: 'net.sf.ehcache', name: 'ehcache', version: '2.7.2'],
+            [group: 'com.github.spullara.mustache.java', name: 'compiler', version: '0.8.12'],
+            [group: 'com.jayway.jsonpath', name: 'json-path', version: '0.9.1'],
+            [group: 'org.apache.tika', name: 'tika-core', version :'1.9'],
+            // Although fineract (at the time of writing) doesn't have any compile time dep. on this,
+            // it's useful to have this for the Spring Boot TestRestTemplate http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#boot-features-rest-templates-test-utility
+            [group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.3.5'],
+            // Once we've switched to Java 8 this dep can be removed.
+            //[group: 'com.google.code.findbugs', name: 'jsr305', version: '3.0.0']
+
+            [group: 'io.swagger', name: 'swagger-jersey-jaxrs', version: '1.5.15'],
+            [group: 'org.apache.activemq', name: 'activemq-broker', version: '5.9.1'],
+            [group: 'javax.validation', name: 'validation-api', version: '2.0.0.Final']
+    )
+
+    implementation 'org.apache.bval:org.apache.bval.bundle:2.0.2'
+
+    testCompile 'junit:junit:4.11',
+            'junit:junit-dep:4.11',
+            'org.mockito:mockito-core:1.9.5',
+            'com.jayway.restassured:rest-assured:2.3.3',
+            [group: 'com.mockrunner', name: 'mockrunner-jms', version: '1.0.6'],
+            [group: 'com.google.code.gson', name: 'gson', version: '2.2.4'],
```

Comment:
I found it slightly surprising that we have to repeat the `api` dependencies here in `testCompile` ... there's probably a better way to do this with Gradle - does anyone know? I'll still merge this though - this always could (should) be cleaned up in a follow-up PR.